2_download/src/fetch_wqp_data.R

```diff
@@ -96,10 +96,13 @@ add_download_groups <- function(sitecounts_df, max_sites = 500, max_results = 25
     purrr::map_dfr(.f = function(df){
       
       df_grouped <- df %>%
+        group_by(CharacteristicName) %>%
         arrange(desc(results_count)) %>%
-        mutate(task_num = MESS::cumsumbinning(x = results_count, 
-                                              threshold = max_results, 
-                                              maxgroupsize = max_sites))
+        mutate(task_num_by_results = MESS::cumsumbinning(x = results_count, 
+                                                         threshold = max_results, 
+                                                         maxgroupsize = max_sites), 
+               task_num = task_num_by_results - 1 + cur_group_id()) %>%
```

Comment:
This is very clever and I had to go in and add `browser()` and compare `task_num_by_results` to `task_num` after this ran to understand it. Maybe a short little comment here? Especially because the comments above this don't mention that groups are then split based on `CharacteristicName`!

Reply:
Yeah, good idea! Do you think the text in https://github.com/USGS-R/ds-pipelines-targets-example-wqp/pull/92/commits/11fa6b342db5a2579d08b2f65d56659798834ea1 helps here?

Reply:
Yes, but I did have to re-read a couple of times to fully understand. Maybe a small example would help?

> For example, both "Specific conductance" and "Temperature" may have values for `task_num_by_results` of 1 and 2 but the values of `cur_group_id()` (1 and 2, respectively) would mean that they have unique values for `task_num` equaling 1, 2, 3, and 4.